app/command.go

```diff
@@ -191,6 +191,100 @@ func (a *App) ExecuteCommand(args *model.CommandArgs) (*model.CommandResponse, *
 	return nil, model.NewAppError("command", "api.command.execute_command.not_found.app_error", map[string]interface{}{"Trigger": trigger}, "", http.StatusNotFound)
 }
 
+type mentionMapItem struct {
+	Name string
+	Id   string
+}
+
+// mentionsToTeamMembers returns all the @ mentions found in message that
+// belong to users in the specified team, linking them to their users
+func (a *App) mentionsToTeamMembers(message, teamId string) model.UserMentionMap {
+	possibleMentions := model.PossibleAtMentions(message)
+	mentionChan := make(chan *mentionMapItem, len(possibleMentions))
+
+	for _, mention := range possibleMentions {
+		go func(mention string) {
+			user, err := a.Srv.Store.User().GetByUsername(mention)
+			if err != nil {
```

Comment:
Let's distinguish a generic error from one that explicitly has a status code of `http.StatusNotFound`.

Reply:
Makes sense, thanks for the suggestion!

Reply:
I was thinking about this and I think that it would make sense to just return outside of the goroutine when `err != nil && err.StatusCode != http.StatusNotFound`. Do you agree, @lieut-data?

Reply:
Agreed: if an underlying error has surfaced, we likely want to signal that back to the caller.

Reply:
Done in https://github.com/mattermost/mattermost-server/pull/13762/commits/552c67e6d459e98b646fd149c792a2bb0eb7486d.